src/wire/packet_manager.cpp

```diff
@@ -319,18 +357,101 @@ void PacketManager::ExecQueryMessage(InputPacket *pkt, const size_t thread_id) {
       std::vector<StatementResult> result;
       std::vector<FieldInfo> tuple_descriptor;
       std::string error_message;
-      int rows_affected;
-
-      // execute the query using tcop
-      auto status = traffic_cop_->ExecuteStatement(
-          query, result, tuple_descriptor, rows_affected, error_message,
-          thread_id);
-
-      // check status
-      if (status == ResultType::FAILURE) {
-        SendErrorResponse(
-            {{NetworkMessageType::HUMAN_READABLE_ERROR, error_message}});
-        break;
+      int rows_affected = 0;
+      std::string query_type;
+
+      std::stringstream stream(query);
+      stream >> query_type;
+
+      if (query_type.compare("PREPARE") == 0) {
+        std::string statement_name;
+        stream >> statement_name;
+        std::size_t pos = query.find("AS");
+        std::string statement_query = query.substr(pos + 3);
+        boost::trim(statement_query);
+
+        // Prepare statement
+        std::shared_ptr<Statement> statement(nullptr);
+
+        LOG_DEBUG("PrepareStatement[%s] => %s", statement_name.c_str(),
+                  statement_query.c_str());
+
+        statement = traffic_cop_->PrepareStatement(statement_name, statement_query,
+                                                   error_message);
+        if (statement.get() == nullptr) {
+          skipped_stmt_ = true;
+          SendErrorResponse(
+              {{NetworkMessageType::HUMAN_READABLE_ERROR, error_message}});
+          LOG_TRACE("ExecQuery Error");
+          return;
+        }
+
+        auto entry = std::make_pair(statement_name, statement);
+        statement_cache_.insert(entry);
+        for (auto table_id : statement->GetReferencedTables()) {
+          table_statement_cache_[table_id].push_back(statement.get());
+        }
+      } else if (query_type.compare("EXECUTE") == 0) {
+        std::string statement_name;
+        std::shared_ptr<Statement> statement;
+        std::vector<type::Value> param_values;
+        bool unnamed = false;
+        std::vector<std::string> tokens;
+
+        boost::split(tokens, query, boost::is_any_of("(), "));
+
+        statement_name = tokens.at(1);
```

Comment:
There are a lot of string operations in this section of code. It would be nice if you could provide more explanations to help people know what this piece of code is doing.